nixos/modules/services/monitoring/telegraf.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.telegraf;
+
+  configFile = pkgs.runCommand "config.toml" {
+    buildInputs = [ pkgs.remarshal ];
+  } ''
+    remarshal -if json -of toml \
+      < ${pkgs.writeText "config.json" (builtins.toJSON cfg.extraConfig)} \
+      > $out
+  '';
+in
+{
+
+  ###### interface
+
+  options = {
+
+    services.telegraf = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable the telegraf server";
+        type = types.bool;
+      };
+
+      package = mkOption {
```

Comment:
I don't understand how packageOverrides would fix my problem. As far as I understand, packageOverride works for the "nix-env" use-case, but it doesn't modify the package that will be used for the NixOS service ?

How can I create a nixos `configuration.nix` that enables a service but uses a different version of the package ?